xbmc/video/guilib/VideoStreamSelectHelper.cpp

```diff
@@ -0,0 +1,423 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "VideoStreamSelectHelper.h"
+
+#include "FileItem.h"
+#include "FileItemList.h"
+#include "ServiceBroker.h"
+#include "application/ApplicationComponents.h"
+#include "application/ApplicationPlayer.h"
+#include "dialogs/GUIDialogSelect.h"
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindowManager.h"
+#include "guilib/LocalizeStrings.h"
+#include "utils/LangCodeExpander.h"
+#include "utils/StreamDetails.h"
+#include "utils/StringUtils.h"
+#include "utils/log.h"
+
+namespace
+{
+constexpr int FILEITEM_INDEX_DISABLE = -2; // FileItem index value to disable the stream
+constexpr int FILEITEM_INDEX_NONE = -1; // FileItem index value for "none" stream
+
+// \brief Make a FileItem entry with "Disable" label, allow to disable a stream
+CFileItemPtr MakeFileItemDisable(const bool isSelected)
+{
+  CFileItemPtr fileItem = std::make_shared<CFileItem>(g_localizeStrings.Get(24021));
+  fileItem->Select(isSelected);
+  fileItem->SetProperty("stream.index", FILEITEM_INDEX_DISABLE);
+  return fileItem;
+}
+
+// \brief Make a FileItem entry with "None" label, signal an empty list
+CFileItemPtr MakeFileItemNone()
+{
+  CFileItemPtr fileItem = std::make_shared<CFileItem>(g_localizeStrings.Get(231));
+  fileItem->SetProperty("stream.index", FILEITEM_INDEX_NONE);
+  return fileItem;
+}
+
+std::shared_ptr<const CFileItem> OpenSelectDialog(CGUIDialogSelect& dialog,
+                                                  const int headingId,
+                                                  const CFileItemList& itemsToDisplay)
+{
+  dialog.Reset();
+  dialog.SetHeading(headingId);
+  dialog.SetUseDetails(true);
+  dialog.SetMultiSelection(false);
+  dialog.SetItems(itemsToDisplay);
+
+  dialog.Open();
+
+  if (dialog.IsConfirmed())
+    return dialog.GetSelectedFileItem();
+
+  return {};
+}
+
+std::string CovertFpsToString(float value)
+{
+  if (value == 0)
```

Comment:
`value == 0.0`- it's a float, you should make that clear.

Reply:
`value == 0.0f` even better

Reply:
That’s what i meant. :-)